revisions.py

```diff
@@ -0,0 +1,482 @@
+            "DATA_ID",
+            "COLL_NAME = '" + rev_data['main_revision_coll'] + "' "
+            "AND META_DATA_ATTR_NAME = '" + originalDataNameKey + "' "
+            "AND META_DATA_ATTR_VALUE = '" + rev_data['main_original_dataname'] + "' ",  # *originalDataName
+            genquery.AS_DICT, ctx)
+
+        for row in iter:
+            log.write(ctx, row['DATA_ID'])
+
+            # based on data id get original_coll_name
+            iter2 = genquery.row_iterator(
+                "META_DATA_ATTR_VALUE",
+                "DATA_ID = '" + row['DATA_ID'] + "' "
+                "AND META_DATA_ATTR_NAME = '" + constants.UUORGMETADATAPREFIX + 'original_path' + "' ",
+                genquery.AS_DICT, ctx)
+            for row2 in iter2:
+                rev_data['original_coll_name'] = row2['META_DATA_ATTR_VALUE']
+
+            rev_data['collection_exists'] = collection.exists(ctx, '/'.join(rev_data['original_coll_name'].split(os.path.sep)[:-1]))
+            rev_data['original_coll_name'] = '/'.join(rev_data['original_coll_name'].split(os.path.sep)[3:])
+
+            # Data is collected on the basis of ORG_COLL_NAME, duplicates can be present
+            try:
+                # This is a double entry and has to be corrected in the total returned to the frontend
+                detail = dict_org_paths[rev_data['original_coll_name']]
+                total = detail[0] + 1
+                dict_org_paths[rev_data['original_coll_name']] = [total, detail[1], detail[2]]
+                # Increment correction as the main total is based on the first query.
+                # This however can have multiple entries which require correction
+                multiple_counted += 1
+            except KeyError:
+                # [count, collect-exists, data-name]
+                dict_org_paths[rev_data['original_coll_name']] = [1, rev_data['collection_exists'], rev['META_DATA_ATTR_VALUE']]
+
+    # create a list from collected data in dict_org_paths
+    revisions = []
+    for key, value in dict_org_paths.items():
+        revisions.append({'main_original_dataname': value[2],
+                          'collection_exists': value[1],
+                          'original_coll_name': key,
+                          'revision_count': value[0]})
+
+    # Alas an extra Query is required to get the total number of rows
+    qtotalrows = Query(ctx, ['COLL_NAME', 'META_DATA_ATTR_VALUE'],
+                       "META_DATA_ATTR_NAME = '" + originalDataNameKey + "' "
+                       "AND META_DATA_ATTR_VALUE like '" + searchString + "%' "
+                       "AND COLL_NAME like '" + startpath + "%' ",
+                       offset=0, limit=None, output=query.AS_DICT)
+
+    # qtotalrows.total_rows() moet worden verminderd met het aantal ontdubbelde entries
+    return {'total': qtotalrows.total_rows() - multiple_counted,
+            'items': revisions}
+
+
+@api.make()
+def api_revisions_list(ctx, path):
```

Comment:
API functions should always be documented, because this is used to generate the API documentation. We use the Sphinx docstring format: https://sphinx-rtd-tutorial.readthedocs.io/en/latest/docstrings.html For examples see https://github.com/UtrechtUniversity/irods-ruleset-uu/blob/8a31c9d2aa98ace579abd47f76211e70208d40b3/folder.py#L102